src/appengine/handlers/cron/group_leader.py

```diff
@@ -62,14 +64,30 @@ def _key_func(testcase):
 
     items = sorted(items, reverse=True, key=lambda t: t.timestamp)
 
-    for item in items:
+    asan_index = None
+    security_index = None
+    i386_indexes = []
+    for idx, item in enumerate(items):
       item.is_leader = False
-
-    leader_index = find_index(items, is_reproducible_and_has_issue)
+      if not security_index and item.security_flag:
+        security_index = idx
+      if not asan_index and item.job_type and '_asan_' in item.job_type:
+        asan_index = idx
+      if item.job_type and environment.is_i386(item.job_type):
+        i386_indexes.append(idx)
+
+    leader_index = security_index
+    leader_index = asan_index
+    if leader_index is None:
+      leader_index = find_index(items, is_reproducible_and_has_issue)
```

Comment:
We still need to keep these prioritisation steps wrt reproducibility and issues. We would also prefer to file a reproducible non-security bug than a non-reproducible security bug. 

One way to keep this is to just sort the `items` list multiple times (`sort()` in python should be stable), first by timestamp, then by a score that we give based on if it's asan/security and its architecture (e.g. asan -> +1, security -> +1, non-x86 -> +1). 

Please also add tests that would've caught this.

Reply:
Fixed the leader selection.